MuvrKitTests/MKWeightPredictorTests.swift

```diff
@@ -0,0 +1,84 @@
+import Foundation
+import XCTest
+@testable import MuvrKit
+
+class MKWeightPredictorTests : XCTestCase {
+    
+    func roundValue(value: Double, forExerciseId exerciseId: MKExercise.Id) -> Double {
+        return MKScalarRounderFunction.roundMinMax(value, minimum: 2.5, step: 2.5, maximum: nil)
+    }
+    
+    func testPeriod() {
+        let predictor = MKPolynomialFittingScalarPredictor(round: roundValue)
+        XCTAssertTrue(predictor.checkPeriod([10.0, 12, 12, 14, 16, 16, 15, 11]))
+        XCTAssertTrue(predictor.checkPeriod([1.0, 2, 3, 4, 5, 6, 7, 7, 6, 2]))
+        XCTAssertFalse(predictor.checkPeriod([15.0, 20, 25, 30, 25, 20, 15, 10]))
+    }
+    
+    func insertSequence<E>(plan: MKExercisePlan<E>, sequence: [E]) {
+        for value in sequence {
+            plan.insert(value)
+        }
+    }
+    
+    func testSequence(sequence: [Double], desiredValue: Double) {
+        let plan = MKExercisePlan<Double>()
+        insertSequence(plan, sequence: sequence)
+        XCTAssertEqual(plan.next.first, desiredValue)
+    }
+    
+    func testPredictWithMarkov() {
+        testSequence([10.0, 15, 20, 25, 30, 35, 30, 30, 15, 20], desiredValue: 25)
+        
+        testSequence([10, 12.5, 15, 17.5, 17.5, 15, 15, 15, 12.5, 12.5, 12.5, 10, 10, 13], desiredValue: 12.5)
+
+        testSequence([10.0, 15, 20, 25, 30, 35, 30, 30, 10.0, 15, 20, 25, 30, 35], desiredValue: 30)
+        
+        testSequence([10.0, 15, 20, 25, 30, 35, 30, 30, 10.0, 15, 20, 25, 30, 35, 30, 30, 10, 10, 10], desiredValue: 10)
+        
```

Comment:
@janm399 from this test, it seems prediction using Markov Chain is pretty good, the new value is selected from the past so there will be no crazy value. This approach is only bad at the first session when we don't have any history sequence.